src/features/saveToList/useSaveObjectToGroup.ts

```diff
@@ -0,0 +1,103 @@
+  const checkedBinding = bindings.checked;
+  const checkedBindingSegments = checkedBinding?.field.split('.');
+  const checkedBindingKey = checkedBindingSegments?.pop();
+
+  const setLeafValue = FD.useSetLeafValue();
+  const { formData } = useDataModelBindings(bindings, DEFAULT_DEBOUNCE_TIMEOUT, 'raw');
+  const appendToList = FD.useAppendToList();
+  const removeFromList = FD.useRemoveIndexFromList();
+
+  const getObjectFromFormDataRow = (row: Row): Row | undefined =>
+    (formData?.group as Row[])?.find((selectedRow) =>
+      Object.keys(row).every((key) => Object.hasOwn(selectedRow, key) && row[key] === selectedRow[key]),
+    );
+
+  const getIndexFromFormDataRow = (row: Row) =>
+    (formData?.group as Row[]).findIndex((selectedRow) => {
+      const { altinnRowId: _ } = selectedRow;
+      return Object.keys(row).every((key) => Object.hasOwn(selectedRow, key) && row[key] === selectedRow[key]);
+    });
+
+  const isRowChecked = (row: Row): boolean => {
+    const formDataObject = getObjectFromFormDataRow(row);
+    if (checkedBindingKey) {
+      return !!formDataObject && !!formDataObject[checkedBindingKey];
+    }
+    return !!formDataObject;
+  };
+
+  const setList = (row: Row) => {
+    if (!bindings.group) {
+      return;
+    }
+    if (isRowChecked(row)) {
+      const index = getIndexFromFormDataRow(row);
+      if (checkedBinding && checkedBindingSegments) {
+        setLeafValue({
+          reference: {
+            ...bindings.checked,
+            field: `${checkedBindingSegments.join('.')}[${index}].${checkedBindingKey}`,
+          } as IDataModelReference,
+          newValue: false,
+        });
+      } else {
+        if (index >= 0) {
+          removeFromList({
+            reference: bindings.group,
+            index,
+          });
+        }
+      }
+    } else {
+      const formDataObject = getObjectFromFormDataRow(row);
+
+      if (formDataObject && checkedBindingKey && checkedBindingKey in formDataObject) {
+        const index = getIndexFromFormDataRow(row);
+        setLeafValue({
+          reference: {
+            ...bindings.checked,
+            field: `${checkedBindingSegments?.join('.')}[${index}].${checkedBindingKey}`,
+          } as IDataModelReference,
+          newValue: true,
+        });
+      } else {
+        const uuid = uuidv4();
+        const next: Row = { [ALTINN_ROW_ID]: uuid };
+        if (checkedBindingKey) {
+          next[checkedBindingKey] = true;
+        }
+        for (const binding of Object.keys(bindings)) {
+          if (binding === 'simpleBinding') {
+            const propertyName = bindings.simpleBinding.field.split('.').pop();
+            if (propertyName) {
+              next[propertyName] = row[propertyName];
+            }
+            //TODO: excluding from List now??
+          } else if (binding !== 'group' && binding !== 'checked' && binding !== 'label' && binding !== 'metadata') {
+            next[binding] = row[binding];
+          }
```

Comment:
Is the TODO here still relevant?

Also, i find this checking for other bindings brittle. When we introduce a new binding to `Checkboxes` or `MultipleSelect`, this will have to be updated (and then possibly be a breaking change for existing `List` component configurations). It will be difficult to notice (TypeScript won't warn us in any way, and no tests will catch this unless configured along with a `group` binding). I would prefer something like:

```ts
if (componentType !== 'List' && binding === 'simpleBinding') {
  ...
} else if (componentType === 'List' && binding !== 'group' && binding !== 'checked') {
  ...
}
```

Also, same goes here, use `dot.str()` to put values in the object when supporting any-depth properties:
```ts
dot.str(relativeBinding, newValue, next);
```

Reply:
This should now be solved with the `useSaveObjectToGroup()` / `useSaveValueToGroup()` hooks.